pkg/nuclide/file-tree/lib/FileTreeController.js

```diff
@@ -737,16 +743,50 @@ class FileTreeController {
       onClose: () => this._closeDialog(),
       shouldSelectBasename: true,
     };
-    this._openDialog(<FileDialogComponent {...props} />);
+    this._openDialog(props);
+  }
+
+  openDuplicateDialog(): void {
+    var selection = this._getSelectedEntryAndDirectoryAndRoot();
+    if (!selection || selection.entry.isDirectory()) {
+      return;
+    }
+
+    var message = (
+      <div>
+        <div>Enter the new name for the file in the root:</div>
+        <div>{path.normalize(selection.root.getPath() + '/')}</div>
+      </div>
+    );
+
+    var {entry, root} = selection;
+
+    var props = {
+      rootDirectory: root,
+      initialEntry: entry,
+      message,
+      onConfirm: async (rootDirectory, relativeFilePath) => {
+        var file = rootDirectory.getFile(relativeFilePath);
+        await file.create();
```

Comment:
Yeah, you are right, didn't think of that. Maybe a [`Notification`](https://atom.io/docs/api/v1.0.1/NotificationManager#instance-addError) can solve that?